dbms/src/Flash/executeQuery.cpp

```diff
@@ -12,27 +12,103 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <Common/FailPoint.h>
+#include <Common/ProfileEvents.h>
+#include <Flash/Coprocessor/DAGContext.h>
 #include <Flash/Coprocessor/DAGQuerySource.h>
 #include <Flash/Planner/PlanQuerySource.h>
 #include <Flash/executeQuery.h>
+#include <Interpreters/Context.h>
+#include <Interpreters/ProcessList.h>
+#include <Interpreters/Quota.h>
 #include <Interpreters/executeQuery.h>
 
+namespace ProfileEvents
+{
+extern const Event Query;
+}
+
 namespace DB
 {
-BlockIO executeQuery(
-    Context & context,
-    bool internal,
-    QueryProcessingStage::Enum stage)
+namespace FailPoints
+{
+extern const char random_interpreter_failpoint[];
+} // namespace FailPoints
+namespace
+{
+void prepareForExecute(Context & context)
+{
+    ProfileEvents::increment(ProfileEvents::Query);
+    context.setQueryContext(context);
+
+    QuotaForIntervals & quota = context.getQuota();
+    quota.addQuery(); /// NOTE Seems that when new time interval has come, first query is not accounted in number of queries.
+    quota.checkExceeded(time(nullptr));
+}
+
+ProcessList::EntryPtr getProcessListEntry(Context & context, const DAGContext & dag_context)
+{
+    if (dag_context.is_mpp_task)
+    {
+        /// for MPPTask, process list entry is created in MPPTask::prepare()
+        RUNTIME_ASSERT(dag_context.getProcessListEntry() != nullptr, "process list entry for MPP task must not be nullptr");
+        return dag_context.getProcessListEntry();
+    }
+    else
+    {
+        RUNTIME_ASSERT(dag_context.getProcessListEntry() == nullptr, "process list entry for non-MPP must be nullptr");
+        return setProcessListElement(
+            context,
+            dag_context.dummy_query_string,
+            dag_context.dummy_ast.get());
+    }
+}
+
+BlockIO executeDAG(IQuerySource & dag, Context & context, bool internal)
```

Comment:
We can remove the `IQuerySource` related code since we don't reuse `executeQuery`?

Reply:
Yes, but here is some relevant code that needs to be cleaned up.
We can remove the IQuerySource in the next pr.

Reply:
In particular, the code related to the `DAGQuerySource`, perhaps we could remove the `IQuerySource` along with the `DAGQueryBlock`.

Reply:
OK